packages/rs-sdk/src/platform/fetch.rs

```diff
@@ -169,26 +155,50 @@ where
         query: Q,
         settings: Option<RequestSettings>,
     ) -> Result<(Option<Self>, ResponseMetadata, Proof), Error> {
-        let request = query.query(sdk.prove())?;
+        let request: &<Self as Fetch>::Request = &query.query(sdk.prove())?;
+
+        let fut = |settings: RequestSettings| {
+            async move {
+                let response = request
+                    .clone()
+                    .execute(sdk, settings)
+                    .await // TODO: We need better way to handle execution response and errors
+                    .map_err(|execution_error| execution_error.into())?;
+
+                let address = response.address.clone();
+                let retries = response.retries;
+                let grpc_response = response.into_inner();
+
+                let object_type = std::any::type_name::<Self>().to_string();
+                tracing::trace!(request = ?request, response = ?grpc_response, object_type, "fetched object from platform");
+
+                let (object, response_metadata, proof): (Option<Self>, ResponseMetadata, Proof) =
+                    sdk.parse_proof_with_metadata_and_proof(request.clone(), grpc_response)
+                        .await
+                        .map_err(|e| ExecutionError {
+                            inner: e,
+                            address: Some(address.clone()),
+                            retries,
+                        })?;
 
-        let response = request
-            .clone()
-            .execute(sdk, settings.unwrap_or_default())
-            .await // TODO: We need better way to handle execution response and errors
-            .map(|execution_response| execution_response.into_inner())
-            .map_err(|execution_error| execution_error.into_inner())?;
+                let o = match object {
+                    Some(item) => Ok((item.into(), response_metadata, proof)),
+                    None => Ok((None, response_metadata, proof)),
+                };
 
-        let object_type = std::any::type_name::<Self>().to_string();
-        tracing::trace!(request = ?request, response = ?response, object_type, "fetched object from platform");
+                o.map(|x| ExecutionResponse {
+                    inner: x,
+                    address,
+                    retries,
+                })
+            }
+        };
 
-        let (object, response_metadata, proof): (Option<Self>, ResponseMetadata, Proof) = sdk
-            .parse_proof_with_metadata_and_proof(request, response)
-            .await?;
+        let settings = sdk
+            .dapi_client_settings
+            .override_by(settings.unwrap_or_default());
 
-        match object {
-            Some(item) => Ok((item.into(), response_metadata, proof)),
-            None => Ok((None, response_metadata, proof)),
-        }
+        retry(settings, fut).await.into_inner()
```

Comment:
So we don't use backon anymore? How do we decide which errors are retriable which are not?